File: problems/002-two-sum/c/brute-force.c
#define solve __glifex_ref_bruteforce
#include "solution.h"

/* O(n^2): check every pair. The "obvious" first approach. */
JVal *solve(JVal *c) {
    JVal *nums = jget(c, "nums");
    double target = jget(c, "target")->num;
    for (int i = 0; i < nums->n; i++)
        for (int j = i + 1; j < nums->n; j++)
            if (nums->items[i]->num + nums->items[j]->num == target) {
                JVal *r = jarr_(2); jpush_(r, jnum_(i)); jpush_(r, jnum_(j)); return r;
            }
    return jarr_(0);
}

File: problems/002-two-sum/rust/brute-force.rs
use crate::json::JVal;

pub fn solve(c: &JVal) -> JVal {
    let nums = c.get("nums").as_arr();
    let target = c.get("target").as_num() as i64;
    // Check every pair -- the obvious first approach, O(n^2).
    for i in 0..nums.len() {
        for j in (i + 1)..nums.len() {
            if nums[i].as_num() as i64 + nums[j].as_num() as i64 == target {
                return JVal::Arr(vec![JVal::Num(i as f64), JVal::Num(j as f64)]);
            }
        }
    }
    JVal::Arr(vec![])
}
